plugin/src/main/java/org/wildfly/plugin/common/StandardOutputStream.java

```diff
@@ -0,0 +1,188 @@
+         */
+        COLLECTING,
+        /**
+         * The output stream data has been discarded.
+         */
+        DISCARDING,
+        /**
+         * The output stream will be redirected to a file.
+         */
+        FILE,
+        /**
+         * The output stream will be redirected to {@link System#err}.
+         */
+        SYSTEM_ERR,
+        /**
+         * The output stream will be redirected to {@link System#out}.
+         */
+        SYSTEM_OUT
+    }
+
+    private final OutputStream delegate;
+    private final Target target;
+    private final Path stdoutPath;
+
+    private StandardOutputStream(final OutputStream delegate, final Target target, final Path stdoutPath) {
+        this.delegate = delegate;
+        this.target = target;
+        this.stdoutPath = stdoutPath;
+    }
+
+    /**
+     * Parses the string and attempts to determine where the data for the stream should be written. The following are
+     * the options for the value:
+     * <ul>
+     * <li>{@code none} indicates the data for this stream will be consumed and {@link #toString()} will return the
+     * data of the {@code discardNone} parameter is {@code false}, otherwise the data will be discarded</li>
+     * <li>{@code System.out} or {@code System.err} to write to the respective stream</li>
+     * <li>Any other value is assumed to be the path to a file and the data will written to the file</li>
+     * </ul>
+     *
+     * @param stdout      the value to be parsed
+     * @param discardNone {@code true} if the {@code stdout} value is {@code none} and the data should be discarded,
+     *                    otherwise the data will be consumed if the {@code stdout} value is {@code none} and will be
+     *                    available via {@link #toString()}
+     *
+     * @return a new output stream
+     *
+     * @throws IOException if there is an error creating the stream
+     */
+    public static StandardOutputStream parse(final String stdout, final boolean discardNone) throws IOException {
+        Assert.checkNotNullParam("stdout", stdout);
+        Path stdoutPath = null;
+        final Target target;
+        final OutputStream out;
+        final String value = stdout.trim().toLowerCase(Locale.ENGLISH);
+        if ("system.out".equals(value)) {
+            out = System.out;
+            target = Target.SYSTEM_OUT;
+        } else if ("system.err".equals(value)) {
+            out = System.err;
+            target = Target.SYSTEM_ERR;
+        } else if ("none".equals(value)) {
+            if (discardNone) {
+                target = Target.DISCARDING;
+                out = ServerProcess.DISCARDING;
+            } else {
+                target = Target.COLLECTING;
+                out = new ByteArrayOutputStream();
+            }
+        } else {
+            target = Target.FILE;
+            // Attempt to create a file
+            stdoutPath = Paths.get(value);
```

Comment:
Only thing I am not 100% about is this. It would make sense to put file under "target",
OTOH it does make sense to have "full path" here.